### PR TITLE
Stop publishing Android Auto extension together with the SDK.

### PR DESCRIPTION
**Pull request by pengdev**
PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).
Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Optimize code for java consumption (`@JvmOverloads`, `@file:JvmName`, etc).
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [x] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [ ] Add an entry inside this element for inclusion in the `mapbox-maps-android` changelog: `<changelog></changelog>`.

### Summary of changes

This PR stop publishing the experimental Android Auto extension together with the SDK.

We will come up with a plan to iterate some plugin/extensions separately and have different versioning for it. The last Android Auto extension release is still available and can be used together with the SDK.

### CODE REVIEW

File: extension-androidauto/build.gradle.kts [hunk 1]
```diff
@@ -50,6 +50,6 @@ project.apply {
   from("$rootDir/gradle/ktlint.gradle")
   from("$rootDir/gradle/lint.gradle")
   from("$rootDir/gradle/jacoco.gradle")
-  from("$rootDir/gradle/sdk-registry.gradle")
+//  from("$rootDir/gradle/sdk-registry.gradle")
```
Comment (alexander-kulikovskii) [inline extension-androidauto/build.gradle.kts:53]: maybe not related to this PR, but you can add some flag `publish-android-auto-extension`

**Comment by kiryldz** [comment]
@pengdev what about creating public ticket about it and adding a link to it in `extension-androidauto/build.gradle.kts`? So that users do not wonder why we have such extension in code but it's not shipped as part of SDK. We could state in the ticket that Android Auto support is for now experimental and that's the reason we don't ship it for now.

**Comment by kiryldz** [comment]
@pengdev also isn't it a breaking change? if someone did use this extension already - what those users should do?

**Comment by tobrun** [comment]
> @pengdev also isn't it a breaking change? if someone did use this extension already - what those users should do?

was never shipped as part of SDK and already published version will remain to work with SDK as-is

**Review by tobrun — APPROVED** [review]

File: extension-androidauto/build.gradle.kts (hunk 1 above)
Comment (tobrun) [reply extension-androidauto/build.gradle.kts:53]: considering this is short lived configuration, I'm fine with commenting it out

**Review by kiryldz — APPROVED** [review]

**Pull request merged**